src/Appwrite/Auth/Hash/Phpass.php

```diff
@@ -0,0 +1,290 @@
+ */
+
+namespace Appwrite\Auth\Hash;
+
+use Appwrite\Auth\Hash;
+
+/*
+ * PHPass accepted options:
+ * int iteration_count_log2; The Logarithmic cost value used when generating hash values indicating the number of rounds used to generate hashes
+ * string portable_hashes
+ * string random_state; The cached random state
+ *
+ * Refference: https://github.com/photodude/phpass
+*/
+class Phpass extends Hash
+{
+    /**
+     * Alphabet used in itoa64 conversions.
+     *
+     * @var    string
+     * @since  0.1.0
+     */
+    protected $itoa64 = './0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz';
+
+    /**
+     * Get default options for specific hashing algo
+     *
+     * @return mixed options named array
+     */
+    public function getDefaultOptions(): mixed
+    {
+        $randomState = \microtime();
+        if (\function_exists('getmypid')) {
+            $randomState .= getmypid();
+        }
+
+        return ['iteration_count_log2' => 8, 'portable_hashes' => false, 'random_state' => $randomState];
+    }
+
+    /**
+     * @param string $password Input password to hash
+     *
+     * @return string hash
+     */
+    public function hash(string $password): string
+    {
+        $options = $this->getDefaultOptions();
+
+        $random = '';
+        if (CRYPT_BLOWFISH === 1 && !$options['portable_hashes']) {
+            $random = $this->getRandomBytes(16, $options);
+            $hash = crypt($password, $this->gensaltBlowfish($random, $options));
+            if (strlen($hash) === 60) {
+                return $hash;
+            }
+        }
+        if (strlen($random) < 6) {
+            $random = $this->getRandomBytes(6, $options);
+        }
+        $hash = $this->cryptPrivate($password, $this->gensaltPrivate($random, $options));
+        if (strlen($hash) === 34) {
+            return $hash;
+        }
+
+        /**
+         * Returning '*' on error is safe here, but would _not_ be safe
+         * in a crypt(3)-like function used _both_ for generating new
+         * hashes and for validating passwords against existing hashes.
+         */
+        return '*';
+    }
+
+    /**
+     * @param string $password Input password to validate
+     * @param string $hash Hash to verify password against
+     *
+     * @return boolean true if password matches hash
+     */
+    public function verify(string $password, string $hash): bool
+    {
+        $hash = $this->cryptPrivate($password, $hash);
+        if ($hash[0] === '*') {
+            $hash = crypt($password, $hash);
+        }
+
+        /**
+         * This is not constant-time.  In order to keep the code simple,
+         * for timing safety we currently rely on the salts being
+         * unpredictable, which they are at least in the non-fallback
+         * cases (that is, when we use /dev/urandom and bcrypt).
+         */
+        return $hash === $hash;
```

Comment:
Shouldnt this be `true`?

Reply:
Woah. Let me fix.. and.. write tests to prevent such a bug.